action.yml

```diff
@@ -45,22 +45,39 @@ inputs:
 outputs:
   needs-github-assistance:
     description: 'Comma-separated list of image names and tags, for image versions that are public and have more than 5000 downloads.'
+    value:  ${{ steps.container-retention-policy.outputs.needs-github-assistance }}
   deleted:
     description: 'Comma-separated list of image names and tags, for image versions that were deleted during the run.'
+    value:  ${{ steps.container-retention-policy.outputs.deleted }}
   failed:
     description: 'Comma-separated list of image names and tags, for image versions that we failed to delete during the run, for an unknown reason.'
+    value:  ${{ steps.container-retention-policy.outputs.failed }}
 runs:
-  using: 'docker'
-  image: 'Dockerfile'
-  args:
-    - ${{ inputs.account-type }}
-    - ${{ inputs.org-name }}
-    - ${{ inputs.image-names }}
-    - ${{ inputs.timestamp-to-use }}
-    - ${{ inputs.cut-off }}
-    - ${{ inputs.token }}
-    - ${{ inputs.untagged-only }}
-    - ${{ inputs.skip-tags }}
-    - ${{ inputs.keep-at-least }}
-    - ${{ inputs.filter-tags }}
-    - ${{ inputs.filter-include-untagged }}
+  using: 'composite'
+  steps:
+    - name: Install dependencies
+      shell: bash
+      run: |
+        # TODO: consider switching to poetry
```

Comment:
We could do this 👍  We would need to do `cat $GITHUB_ACTION_PATH/poetry.lock > poetry.lock` or something first, to load the lockfile into the context of the Github runner, but I think that should work.

I didn't know about `$GITHUB_ACTION_PATH` when I wrote this 🙃
```suggestion
```

Reply:
I think it's better suited for a separate PR. I'll leave it as is for now.
Maybe someone else would like to tackle that :)

Reply:
Anyway, redirecting lock file to current working dir is a bad idea since it may overwrite the one in there.
Haven't used poetry yet, so not sure if it can take lock file as a parameter.
Maybe simply changing `working-directory` for poetry install would be enough.